PG-autoloads-bootstrap.el

```diff
@@ -0,0 +1,18 @@
+;;;###autoload
+;;; Bootstrap ProofGeneral's autoloads.
+;; The code in package.el will not look in sub-directories of "the package directory"
+;; \(c.f. `package-generate-autoloads')
+;; , so it needs a little help from the code below to find ProofGeneral's autoloads.
+(let* ((base-dir (file-name-directory
+                  (if load-in-progress load-file-name
+                    buffer-file-name)))
+       (autoloads-addendum (expand-file-name "PG-autoloads-addendum.el" base-dir)))
+  (unless (file-exists-p autoloads-addendum)
+    (let ((generated-autoload-file autoloads-addendum))
+      (dolist (name (directory-files-recursively base-dir "" t))
+        (when (file-directory-p name)
+          (update-directory-autoloads name)))))
+  ;; the load below is a wart, but it doesn't look I get full functionality if I just rely on autolads
```

Comment:
It's a verbatim copy of the code you had [submitted to real-gud](https://github.com/realgud/realgud/issues/84).  It  walks the sub-directories of the directory in which this file is located, and adds the autoloads  found in each sub-directory to the `generated-autoload-file` (the call to `update-directory-autoloads` is in the dynamic scope of the let-binding, so the let-binding of `generated-autoload-file` is visible to the callee).

Or were you referring to something else?